index.js

```diff
@@ -81,6 +81,43 @@ rbush3d.prototype = {
         return false;
     },
 
+    raycastInv: function (ox, oy, oz, idx, idy, idz, length) {
+        length = length || Infinity;
+        var node = this.data,
+            toBBox = this.toBBox,
+            result,
+            dist = Infinity;
+
+        if (idx === Infinity && idy === Infinity && idz === Infinity) return undefined;
+        if (boxRayIntersects(node, ox, oy, oz, idx, idy, idz) === Infinity) return undefined;
+
+        var nodesToSearch = [],
+            i, len, child, childBBox;
+
+        while (node) {
+            for (i = 0, len = node.children.length; i < len; i++) {
+                child = node.children[i];
+                childBBox = node.leaf ? toBBox(child) : child;
+                var d = boxRayIntersects(childBBox, ox, oy, oz, idx, idy, idz);
+                if (d < length) {
+                    if (!node.leaf) {
+                        nodesToSearch.push(child);
+                    } else if (d < dist) {
+                        dist = d;
+                        result = child;
+                    }
+                }
+            }
+            node = nodesToSearch.pop();
+        }
+
+        return result;
+    },
+
+    raycast: function (ox, oy, oz, dx, dy, dz, length) {
+        return this.raycastInv(ox, oy, oz, 1 / dx, 1 / dy, 1 / dz, length);
```

Comment:
Check for degenerate intersection if dx,dy,dz is 0.

Reply:
already check `Infinity` in `raycastInv` and `1 / 0` is `Infinity`